packages/go_router/lib/src/router.dart

```diff
@@ -186,6 +183,29 @@ class GoRouter extends ChangeNotifier implements RouterConfig<RouteMatchList> {
         RouteInformation(location: location, state: extra);
   }
 
+  /// Navigate to the last location of the [StatefulShellBranch] at the provided
+  /// index in the associated [StatefulShellBranch].
+  ///
+  /// If the branch has not been visited before, this method will navigate to
+  /// initial location of the branch. Consider using
+  /// [StatefulShellRouteState.goBranch] as a more convenient alternative to
+  /// this method.
+  void goBranch(int index, StatefulShellRouteState shellState) {
```

Comment:
I still feel a bit uncertain about adding this method in GoRouter class. I think we should add this in StatefulNavigationShell

Reply:
Yeah, I know, I felt that as well at first too. But the upside is that that your get access to the internals of GoRouter in a convenient way. And in the preload branch I mentioned above ([RouteMatcher](https://github.com/tolo/flutter_packages/tree/statefulshellroute-preload)), I've converted the goBranch method into a more generic version (goShellNavigator). Please have a look and see what you think of that solution.

Reply:
I think I would rather expose necessay API in GoRouter or other classes that are necessary to implement the goBranch in StatefulnavigationShell. Even though this method help hide some internal API in goRouter, but this method itself is not a really useful API as well....